app/scripts/directives/addConfigToApplication.js

```diff
@@ -25,6 +25,26 @@
   function AddConfigToApplication($filter, $scope, APIService, ApplicationsService, DataService, Navigate, NotificationsService, StorageService) {
     var ctrl = this;
     var humanizeKind = $filter('humanizeKind');
+    ctrl.canAddRefToApplication = false;
+
+    var checkContainerForRef = function(container) {
+      var addRefName = ctrl.apiObject.metadata.name;
+      if (ctrl.apiObject.kind === "ConfigMap") {
+        return _.find(container.envFrom, {configMapRef: {name: addRefName}});
+      } else {
+        return _.find(container.envFrom, {secretRef: {name: addRefName}});
+      }
+    };
+
+    ctrl.checkApplicationContainersRefs = function(application) {
+      ctrl.canAddRefToApplication = false;
+      _.each(_.get(application, 'spec.template.spec.containers'), function(container) {
+        if (!checkContainerForRef(container)) {
+          ctrl.canAddRefToApplication = true;
+          return false;
+        }
+      });
```

Comment:
I think this could be shortened to.

```js
var containers = _.get(application, 'spec.template.spec.containers');
ctrl.canAddRefToApplication = !_.every(containers, checkContainerForRef);
```